Docs-5685-partner-manager-EA-mc (#33)
mulesoft/docs-partner-manager: modules/ROOT/pages/activity-tracking.adoc

include::_attributes.adoc[]
endif::[]

B2B transactions often communicate contractual obligations between companies.
It is desirable (and often necessary) to track transmissions that are sent or received and store the transmission content as well as any metadata.
Metadata can be the date and time the transmission was sent or received, the identity of the partner, and the success or failure of processing.
Some uses for this data include:

* Non-repudiation.
* Correlation of acknowledgments.
* Error correction.
* Auditing.

> Anypoint Partner Manager enables you to monitor B2B activity tracking for transmissions.

Reviewer: ```suggestion
Anypoint Partner Manager enables you to monitor B2B transmission activity.
```

> Activity is the full set of end-to-end events (including transmissions, documents, and errors) involved in processing a transmission.

Reviewer: ```suggestion
Activity is the full set of end-to-end events (including transmission meta data, messages, and errors) involved in processing a transmission.
```

You can:

> * Search for document metadata (such as filename, control number, endpoint, or date received).

Reviewer: ```suggestion
* Search for transmission metadata (such as partner name, message type, control number, or date received).
```

Author: The UI shows filename, not partner name.

> * Filter to find the specific transmission flow data.

Reviewer: Can't do this yet. Remove

Author: Commented out.

> * View and download the payload information from the receive transmission.

Reviewer: ```suggestion
* View and download the payload information from the transmission.
```

> * Review and monitor errors and acknowledgements.

Reviewer: ```suggestion
* Review errors and acknowledgement payloads.
```

== Track Activity for Message Flows:

. Open Partner Manager.
. From within your Partner Manager environment, select Activity from the left-hand navigation. You see the activity on the message flows you created.

> . Scroll the list. If you don't see the desired message flow, type a filename, control number, endpoint, or date received in the search box.

Reviewer: ```suggestion
. Scroll the list. If you don't see the desired transmission, type a partner name, message type, control number, or date received in the search box.
```

Author: Fixed. You can also search on filename so I have included it here as well.

> . Select the message flow. You see a summary with the File name, Control Number, Endpoint, and date and time the transmission was received.

Reviewer: ```suggestion
. Select the transmission. You see a summary with the File name, Control Number, Endpoint, and date and time the transmission was received.
```

The Transmission Flow section shows the stages of the flow. If you see an exclamation point next to one of the stages it means that this is the place where there is an issue with the activity.
. Click the *View Payload* button to see the payload transmission.
. Click the *Download* button to download the payload for analysis.

> . Continue for any other message flow.

Reviewer: ```suggestion
. Continue for any other transmission.
```

> include::_attributes.adoc[]

Reviewer 2: Suggest rewording title to Track B2B Transmission Activity.

Author: I'm not sure that's accurate.

Author: I'll need to run it by the PM and see what he says.

endif::[]

B2B transactions often communicate contractual obligations between companies.

> It is desirable (and often necessary) to track transmissions that are sent or received and store the transmission content as well as any metadata.

Reviewer 2: ```suggestion
Due to the sensitivity of this information, it is desirable (and often necessary) to track transmissions and store both transmission content and metadata.
```

> Metadata can be the date and time the transmission was sent or received, the identity of the partner, and the success or failure of processing.

Reviewer 2: ```suggestion
Metadata includes things like the date and time the transmission was sent or received, the identity of the partner, and the success or failure of processing.
```

> Some uses for this data include:

Reviewer 2: ```suggestion
Uses for this data include:
```
(You don't need "some" when you use "include.")

> * Non-repudiation.

Reviewer 2: Nonrepudiation

No hyphen, no period. GLOBALLY avoid hyphenating prefixes, per the style reference.

Reviewer 2: Remove periods from each list item; they are not sentences.

Author: Will do throughout.

Author: Fixed.

* Correlation of acknowledgments.
* Error correction.
* Auditing.

> Anypoint Partner Manager allows you to monitor B2B transmission activity.

Reviewer 2: ```suggestion
Anypoint Partner Manager enables you to monitor B2B transmission activity, including message metadata, message content, and errors.
```

Author: He wil say no because he wants the full definition here.

> Activity is the full set of end-to-end events involved in processing a transmission. This includes transmission metadata, messages, and errors.

Reviewer 2: omit

Author: See previous.

> You can:

Reviewer 2: You can perform the following transmission monitoring tasks:

Reviewer 2: Remove periods from list items below.

Author: Great, thanks.

* Search for document metadata (such as partner name, message type, filename, control number, endpoint, or date received).
* View and download the payload information from the transmission.
* Review errors and acknowledgement payloads.
// * Filter to find the specific transmission flow data.

> == Track Activity for Message Flows:

Reviewer 2: Replace with plain text, because this is the real topic, not a subtopic:

     To track B2B transmission activity:

Author: okay.

Author: I was told that we were to remove the 'To do... Is that correct?

. Open Partner Manager.

> . From within your Partner Manager environment, select Activity from the left-hand navigation. You see the activity on the message flows you created.

Reviewer 2: ```suggestion
. Select *Activity* from the navigation pane.
```

Reviewer 2: The first step already locates the user. The second sentence merely confirms the intuitive result.
GLOBALLY minimize task steps to one user action apiece. Avoid documenting expected results and displays.

Author: Got it, thanks.

> . Scroll the list. If you don't see the desired transmission then type a partner name, message type, filename, control number, endpoint, or date received in the search box.

Reviewer 2: ```suggestion
. Scroll the list for the transmission you are interested in. 
If you do not see that transmission, use the search box to find a partner name, message type, filename, control number, endpoint, or date received.
```

> . Select the transmission. You see a summary with the File name, Control Number, Endpoint, and date and time the transmission was received.

Reviewer 2: ```suggestion
. Select the transmission and review the summary information.
```

> The Transmission Flow section shows the stages of the flow including any acknowledgement you set. If you see an exclamation point in the rectangle box next to one of the stages it means that this is the place where there is an issue with the transmission. Typically, any errors can be viewed in the transmission payload. To inspect the payload:

Reviewer 2: Turn this into a step:

```suggestion
. Scan the Transmission Flow section for an exclamation point (!) in any of the rectangle boxes next to the displayed flow stages, indicating an issue with the transmission.
```

> . Click the *View Payload* button to see the payload transmission.

Reviewer 2: ```suggestion
. If you see an exclamation point, click *View Payload > Download* to download the payload and troubleshoot the source of the error.
```

Author: Can't really combine these steps due to the fact that users may want to just preview the payload and not download it. Also, it's not a menu selection.

> . Click the *Download* button to download the payload for analysis.

Reviewer 2: Omit per previous.

Author: Oh, see you fixed it! Ignore my previous comment.

> . Continue for any other transmission.

Reviewer 2: Suggest omitting this. I think it is understood that if you have more than one to check, you repeat the steps.
An alternative is also to keep this, but make it plain text following the last step: Repeat these steps for every transmission that you want to monitor.
If you keep this as a step...:
```suggestion
. Repeat steps to monitor another transmission.
```

Author: Like the first suggestion.

> // Refer to the Troubleshooting section for common message flow errors.

Reviewer 2: Consider replacing with See Also link.

Author: It's commented out because I don't have any TS info yet. Hoping to get some in the next few days, but may not happen. I will move it anyway.
